packages/commercetools/api-client/src/index.server.ts

```diff
@@ -70,13 +70,23 @@ const tokenExtension: ApiClientExtension = {
         auth: {
           onTokenChange: (newToken) => {
             if (!currentToken || currentToken.access_token !== newToken.access_token) {
-              res.cookie('vsf-commercetools-token', JSON.stringify(newToken));
+              res.cookie(
+                'vsf-commercetools-token',
+                JSON.stringify(newToken),
+                { expires: newToken?.expires_at ? new Date(newToken.expires_at) : new Date() }
```

Comment:
Maybe?:
```ts
{ expires: new Date(newToken?.expires_at || null) }
```

Reply:
I pushed a fix. Setting date to the past (which will be a case if we do `new Date(null)` removes a cookie, which is not intended behavior.